openspoor/utils/saferequests.py

```diff
@@ -0,0 +1,91 @@
+import time
+import json
+import certifi
+import urllib3
+from typing import Optional
+
+from openspoor.utils.singleton import Singleton
+
+
+class SafeRequest(Singleton):
+
+    last_request = 0  # Use a class attribute, as we use the Singleton pattern
+
+    def __init__(self, max_retry: int = 5, time_between: float = 1.0):
+        """
+        Create a package from where we can make requests. The requests are done using certificates and a sleep is
+        built in to guarantee that there is some time between every API call.
+
+        :param max_retry: The maximum amount of times a request is attempted
+        :param time_between: The minimum time between two consecutive queries
+        """
+        self.pool = urllib3.PoolManager(ca_certs=certifi.where())
+        self.max_retry = max_retry
+        self.time_between = time_between
+
+    def force_time_between_and_retry(func):
```

Comment:
curious as why you chose a wrapper design pattern here, instead of just making retryer the _request_with_retry function that just calls self._retry and use that in the rest of the class.

Reply:
Agreed, the decorator with wrapper is a bit much here. Everything is in just one function now.